MvvmCross.Droid.Support.V7.RecyclerView/MvxRecyclerView.cs

```diff
@@ -44,6 +46,32 @@ public MvxRecyclerView(Context context, IAttributeSet attrs, int defStyle, IMvxR
                 ItemTemplateId = itemTemplateId;
         }
 
+        public override void OnStartTemporaryDetach()
+        {
+            _temporarilyDetached = true;
+            base.OnStartTemporaryDetach();
+        }
+
+        public override void OnFinishTemporaryDetach()
+        {
+            _temporarilyDetached = false;
+            base.OnFinishTemporaryDetach();
+        }
+
+        protected override void OnDetachedFromWindow()
+        {
+            base.OnDetachedFromWindow();
+
+            if (!_temporarilyDetached)
+            {
+                // Clear out all bindings from the ViewHolders created by the current adapter.
+                // This is for https://github.com/MvvmCross/MvvmCross/issues/1379
+                // According to the documentation Adapters can be owned by multiple
+                // RecyclerViews so this might not be entirely accurate.
+                this.Adapter?.ClearAllBindings();
```

Comment:
Just to clarify: this is the `RecyclerView` calling the adapter, not the `ViewHolder` calling the adapter is that what you meant?

Unfortunately just listening for the detach event on the `ViewHolder` doesn't help us because the `ViewHolder` can be recycled (this was my original approach).  If you clear the `BindingContext` when the `ViewHolder` detaches then bindings will fail to work when the `ViewHolder` is reattached.